### ISSUE TITLE
Download as .py yields strange file

### ISSUE THREAD

**Issue by d3ld0t** [ipython/ipython#10731]
Using jupyter 4.3.0 python27 installed with conda. I have updated to the latest version.

Whenever I download any .py file, all I get is a .py file with the following:

```
talk 1
talk 2
talk 3
talk 4
talk 5
```

and nothing else. This is consistent across machines running from the same network mounted jupyter directory. Other versions on other machines do not have this issue.

Any ideas?

**Comment by takluyver** [comment]
Are you downloading `.py` files that already exist on the server, or downloading a notebook as a .py file? What's in the notebook?

**Comment by d3ld0t** [comment]
I should have clarified. I run many python notebooks (.ipynb files) and it is these that I am trying to download as a .py file.  The notebook files exist already and downloading any of these as .py results in the aforementioned output. Downloading as .html or as .ipynb works fine.

I should note that this seems like an issue with nbconvert as the same thing happens when I try to convert the .ipynb into a .py via "jupyter nbconvert --to script foo.ipynb"

**Comment by takluyver** [comment]
Is it possible that you have a file called `python.tpl` in that directory? If so, nbconvert might be trying to use it as a template.

**Comment by d3ld0t** [comment]
There is no such file in the directory where I am trying to save files from. In addition, I searched for all files with the extension .tpl in my home directory (where conda lives as well as all jupyter and nbconvert packages) and found nothing.

**Comment by d3ld0t** [comment]
I just created a simple template file with the following:

```
## remove markdown cells
{% block markdowncell -%}
{% endblock markdowncell %}

## change the appearance of execution count
{% block in_prompt %}
# [{{ cell.execution_count if cell.execution_count else ' ' }}]:
{% endblock in_prompt %}
```

I then do:

`jupyter nbconvert --to script foo.ipynb --stdout --template=simplepython.tpl
`

and get the same output:

```
[NbConvertApp] Converting notebook foo.ipynb to script
talk 1
talk 2
talk 3
talk 4
talk 5
```

**Comment by takluyver** [comment]
If you look at the notebook metadata, is there a field `language_info.nbconvert_exporter`? What is its value?

The value should normally be `'python'`. if it is, can you try this:

```python
import entrypoints
entrypoints.get_single('nbconvert.exporters', 'python')
```

Also, check the `language_info.name` field in the notebook, and try:

```python
import entrypoints
entrypoints.get_single('nbconvert.exporters.script', 'python')
```

**Comment by d3ld0t** [comment]
nbconvert_exporter is indeed python:

```
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.13"
  }
 },

```

Running
`import entrypoints
entrypoints.get_single('nbconvert.exporters', 'python')`

yields

`EntryPoint('python', u'nbconvert.exporters', u'PythonExporter', Distribution('nbconvert', '5.2.1-py2.7'))
In [ ]:
`

and running 
`import entrypoints
entrypoints.get_single('nbconvert.exporters.script', 'python')`

raises a NoSuchEntryPoint exception:
```
---------------------------------------------------------------------------
NoSuchEntryPoint                          Traceback (most recent call last)
<ipython-input-82-ad747417dff7> in <module>()
      1 import entrypoints
----> 2 entrypoints.get_single('nbconvert.exporters.script', 'python')

/home/anuj/miniconda2/lib/python2.7/site-packages/entrypoints.pyc in get_single(group, name, path)
    184                 return EntryPoint.from_string(epstr, name, distro)
    185 
--> 186     raise NoSuchEntryPoint(group, name)
    187 
    188 def get_group_named(group, path=None):

NoSuchEntryPoint: No 'python' entry point found in group 'nbconvert.exporters.script'

```

**Comment by takluyver** [comment]
That all sounds normal.

I just thought of another possibility on the template: is there a file called `python` with no extension in that directory?

**Comment by d3ld0t** [comment]
Yes!! There was a file in my home directory with that exact text! I removed it and now it works!
Thanks so much!!!!

**Comment by takluyver** [comment]
We should probably make it a bit more robust against that.

**Comment by takluyver** [comment]
I've made a pull request jupyter/nbconvert#639 to make this a bit harder to hit accidentally.
